### PR TITLE
[libpq, openssl, librtmp] libpq 12 and others with openssl 1.1.1d

### PR DESCRIPTION
**Pull request by Neumann-A**
Lets see how far CI gets with this PR and push the openssl upgrade a bit

### CODE REVIEW

**Comment by Neumann-A** [comment]
So most errors found so far:
windows-static: Missing linkage to crypt32.lib or some other
linux: Missing linkage to pthread.

Most common point of failure. Detection logic for openssl 1.1 fails due to the one above......

remaining linux issues: Probably missing pthread either in the detection or the later link stage

**Comment by Neumann-A** [comment]
Concerning the UWP build error. 
I only took a look @ quickfix and the problem is: Openssl defines PEM_read_X509 as:
`PEM_read_X509                           1287	1_1_0	EXIST::FUNCTION:STDIO`
and the UWP patch does:
`        disable         => sub { [ 'ui-console', 'stdio', 'async', 'uplink',`

Since OpenSSL master has the same target definition and stdio is disabled, I have to assume that the missing Symbols should have never been there in the first place and so that the UWP regressions are false negatives.

**Comment by Neumann-A** [comment]
Yeah linux is green. now just 16h more for osx to decide what it wants

**Comment by JackBoosY** [comment]
Hi @Neumann-A, seems there are some conflicts, please merge master into this branch,

Thanks.

**Comment by JackBoosY** [comment]
|arm-uwp             |test|notes
|--------------------|----|-----
|offscale-libetcd-cpp|Skip|Missing port dependency
|grpc                |Fail|**Regression**
|googleapis          |Skip|Missing port dependency
|xmlsec              |Fail|**Regression**
|gsoap               |Pass|

|x64-uwp |test|notes
|--------|----|-----
|libpq   |Skip|Missing port dependency
|quickfix|Fail|**Regression**

**Comment by Neumann-A** [comment]
@JackBoosY 
grpc error:
`       C:\vsts\_work\2\s\buildtrees\grpc\src\v1.23.1-ebfd5c51df\src\core\tsi\alts\crypt\aes_gcm.cc(85,5): error C3861:  'snprintf': identifier not found [C:\vsts\_work\2\s\buildtrees\grpc\arm-uwp-dbg\grpc.vcxproj]`
seems unrelated to this PR. (Could be that OpenSSL was wrongly pulling in snprintf in UWP before)

xmlsec error:
`         x509.obj : error LNK2019: unresolved external symbol BN_print_fp referenced in function xmlSecOpenSSLX509CertDebugDump [C:\vsts\_work\2\s\buildtrees\xmlsec\arm-uwp-dbg\libxmlsec-openssl.vcxproj]`
BN_print_fp requires stdio which is marked as not available in UWP builds of OpenSSL. As such it is a false positiv and should have not worked before

quickfix errors:
 ```
        C:\vsts\_work\2\s\buildtrees\quickfix\src\v1.15.1-5086b0da20\src\C++\UtilitySSL.cpp(713,5): error C3861:  'ERR_print_errors_fp': identifier not found [C:\vsts\_work\2\s\buildtrees\quickfix\x64-uwp-dbg\src\C++\quickfix.vcxproj]
         C:\vsts\_work\2\s\buildtrees\quickfix\src\v1.15.1-5086b0da20\src\C++\UtilitySSL.cpp(866,8): error C3861:  'PEM_read_X509': identifier not found [C:\vsts\_work\2\s\buildtrees\quickfix\x64-uwp-dbg\src\C++\quickfix.vcxproj]
         C:\vsts\_work\2\s\buildtrees\quickfix\src\v1.15.1-5086b0da20\src\C++\UtilitySSL.cpp(909,8): error C3861:  'PEM_read_PrivateKey': identifier not found [C:\vsts\_work\2\s\buildtrees\quickfix\x64-uwp-dbg\src\C++\quickfix.vcxproj]
``` 
basically the same as before. OpenSSL marks those functions as requiring stdio which OpenSSL say is not available. As such this is also a false positiv and it should have not worked before. 

see also https://github.com/microsoft/vcpkg/pull/8566#issuecomment-541460751

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1282,6 +1283,7 @@ pbc:x64-osx=fail
 pcre2:arm-uwp=fail
 pdal:x64-linux=fail
 pdal:x64-osx=fail
+pdal:x64-windows-static=fail
```
Comment (Neumann-A) [inline scripts/ci.baseline.txt]: pdal requires probably the same fix as gdal

e.g. linking against:
https://github.com/microsoft/vcpkg/pull/8566/commits/206b7d4a0f55ea489fb16a3a7dd7747cb176b188
https://github.com/microsoft/vcpkg/pull/8566/commits/238c9cb2110349d48cf482bfc58489f9b639c66a

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -465,6 +465,7 @@ freeglut:x64-uwp=fail
 freeimage:x64-osx=fail
 freerdp:arm64-windows=fail
 freerdp:arm-uwp=fail
+freerdp:x64-linux=fail
```
Comment (Neumann-A) [inline scripts/ci.baseline.txt]: Seems to be missing Threads::Threads linkage in the CMakeLists.txt

**Comment by vicroms** [comment]
Passing all CI checks.
Thanks for your excellent work @Neumann-A!

**Comment by Neumann-A** [comment]
@vicroms you should thank the others too.
